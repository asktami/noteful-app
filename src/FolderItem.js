import React from 'react';

import NotefulContext from './NotefulContext';

const FolderItem = props => {
	const { folders } = this.context;
	const { notes } = this.context;

	// use the selected noteId to
	// - use the notes object & get the folderId
	// - from the folders object, use the folderId to get the note's parent folder

	const activeNote = notes
		? notes.find(note => note.id == props.match.params.noteId)
		: '';
	const folderId = activeNote.folderId;
	const activeFolder = folders.filter(folder => folder.id == folderId);

	return (
		<NotefulContext.Consumer>
			{context => (
				<>
					{!activeFolder
						? null
						: activeFolder.map(folder => (
								<div className="header-container" key={folder.id}>
									<span>
										<h2>{folder.name}</h2>
									</span>
								</div>
						  ))}

					<br />
					<button className="btn-save" onClick={props.history.goBack}>
						Go Back
					</button>
				</>
			)}
		</NotefulContext.Consumer>
	);
};

export default FolderItem;
