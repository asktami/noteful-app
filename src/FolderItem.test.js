import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import FolderItem from './FolderItem';

describe(`FolderItem component`, () => {
	const folders = [
		{
			id: 101,
			label: 'test-folder-label'
		}
	];

	const notes = [
		{
			id: 101,
			folderId: 101,
			title: 'test-note-title',
			modified: new Date(2019, 8, 30)
		}
	];

	const props = {
		match: {
			params: {
				noteId: 101
			}
		},
		history: {
			goBack: null
		}
	};

	it('renders a FolderItem by default', () => {
		const wrapper = shallow(
			<FolderItem {...props} folders={folders} notes={notes} />
		);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('renders the FolderItem given props', () => {
		const wrapper = shallow(
			<FolderItem {...props} folders={folders} notes={notes} />
		);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
