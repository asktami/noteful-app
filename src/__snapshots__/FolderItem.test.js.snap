// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FolderItem component renders a FolderItem by default 1`] = `
<Fragment>
  <div
    className="header-container"
    key="101"
  >
    <span>
      <h2>
        test-folder-label
      </h2>
    </span>
  </div>
  <br />
  <button
    className="btn-save"
    onClick={null}
  >
    Go Back
  </button>
</Fragment>
`;

exports[`FolderItem component renders the FolderItem given props 1`] = `
<Fragment>
  <div
    className="header-container"
    key="101"
  >
    <span>
      <h2>
        test-folder-label
      </h2>
    </span>
  </div>
  <br />
  <button
    className="btn-save"
    onClick={null}
  >
    Go Back
  </button>
</Fragment>
`;
